database-commons/src/main/java/io/cdap/plugin/db/sink/AbstractDBSink.java

```diff
@@ -438,23 +429,24 @@ private void validateSchema(FailureCollector collector, Class<? extends Driver>
 
     Properties connectionProperties = new Properties();
     connectionProperties.putAll(dbSinkConfig.getConnectionArguments());
-    try (Connection connection = DriverManager.getConnection(connectionString, connectionProperties)) {
+    try (Connection connection = RetryUtils.createConnectionWithRetry((RetryPolicy<Connection>) retryPolicy,
+      connectionString, connectionProperties, getErrorDetailsProvider())) {
       executeInitQueries(connection, dbSinkConfig.getInitQueries());
       try (ResultSet tables = connection.getMetaData().getTables(null, dbSchemaName, tableName, null)) {
         if (!tables.next()) {
-          collector.addFailure(
-            String.format("Table '%s' does not exist.", tableName),
-                          String.format("Ensure table '%s' is set correctly and that the connection string '%s' " +
-                                  "points to a valid database.", fullyQualifiedTableName, connectionString))
+          collector.addFailure(String.format("Table '%s' does not exist.", tableName),
+            String.format("Ensure table '%s' is set correctly and that the connection string '%s' " +
+              "points to a valid database.", fullyQualifiedTableName, connectionString))
             .withConfigProperty(DBSinkConfig.TABLE_NAME);
           return;
         }
       }
       setColumnsInfo(inputSchema.getFields());
-      try (PreparedStatement pStmt = connection.prepareStatement(String.format("SELECT %s FROM %s WHERE 1 = 0",
-                                                                               dbColumns,
-                                                                               fullyQualifiedTableName));
-           ResultSet rs = pStmt.executeQuery()) {
+      try (PreparedStatement pStmt = RetryUtils.prepareStatementWithRetry((RetryPolicy<PreparedStatement>) retryPolicy,
+        connection, String.format("SELECT %s FROM %s WHERE 1 = 0", dbColumns, fullyQualifiedTableName),
+          getErrorDetailsProvider());
+           ResultSet rs = RetryUtils.executeQueryWithRetry((RetryPolicy<ResultSet>) retryPolicy, pStmt,
+             getErrorDetailsProvider())) {
         getFieldsValidator().validateFields(inputSchema, rs, collector);
       }
     } catch (SQLException e) {
```

Comment:
I don't think this catch is needed now?

Reply:
Both the places where ever we are connecting to SQL driver, I can see RetryUtils getting used, how will it be used?

Reply:
Added c4469b210f499a1f8cdddb51af87a8e7aa9412dd